src/server/migrations/migrateDatabase.js

```diff
@@ -0,0 +1,191 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const db = require('../models/database').db;
+const Migration = require('../models/Migration');
+const { compareSemanticVersion } = require('../util');
+const { log } = require('../log');
+
+// file needed to run database transaction
+const requiredFile = [];
+
+/**
+ * @param migrationItems extracted from registerMigration.js
+ * @returns {Array} of migration pair
+ */
+function printMigrationList(migrationItems) {
+	const list = [];
+	for (const item of migrationItems) {
+		list.push(`${item.fromVersion} -> ${item.toVersion}`);
+	}
+	return list;
+}
+
+/**
+ * Create an array of unique versions;
+ * @param migrationItems extracted from registerMigration.js
+ * @returns {*[]} array of unique versions
+ */
+function getUniqueKeyOfMigrationList(migrationItems) {
+	const vertex = [];
+	for (const m of migrationItems) {
+		// disallow down migration
+		if (compareSemanticVersion(m.fromVersion, m.toVersion) === 1) {
+			throw new Error(`Migration fromVersion ${m.fromVersion} is more recent than toVersion ${m.toVersion}`);
+		} else {
+			vertex.push(m.fromVersion);
+			vertex.push(m.toVersion);
+		}
+	}
+
+	return [...new Set(vertex)];
+}
+
+/**
+ * create an adjacency list (OBJECT) of the migrations
+ * @param migrationItems extracted from registerMigration.js
+ * @returns {{}}
+ */
+function createMigrationList(migrationItems) {
+	const migrationList = {};
+
+	const uniqueKey = getUniqueKeyOfMigrationList(migrationItems);
+
+	uniqueKey.forEach(key => {
+		migrationList[key] = [];
+	});
+
+	for (const m of migrationItems) {
+		migrationList[m.fromVersion].push(m);
+	}
+
+	return migrationList;
+}
+
+/**
+ * If current version or version user wants to migrate is not in the migrationList, throw an Error
+ * @param curr current version of the database
+ * @param to version want to migrate to
+ * @param adjListArray adjacency list of version graph
+ */
+function checkIfFromAndToExist(curr, to, adjListArray) {
```

Comment:
I like using names more like `ensure` and `assert` for functions that throw an error if a condition isn't validated. For me, `check` implies that it's going to return a boolean to signal failure. You don't need to change this if you don't agree though.